phoenix-core/src/main/java/org/apache/phoenix/compile/DeleteCompiler.java

```diff
@@ -486,19 +494,22 @@ else if (table.isTransactional() && connection.getSCN() != null) {
             projectedColumns.add(column);
             aliasedNodes.add(FACTORY.aliasedNode(null, FACTORY.column(null, '"' + column.getName().getString() + '"', null)));
         }
-        // Project all non PK indexed columns so that we can do the proper index maintenance
+        // Project all non PK indexed columns so that we can do the proper index maintenance on the indexes for which
+        // mutations are generated on the client side. Indexed columns are needed to identify index rows to be deleted
         for (PTable index : table.getIndexes()) {
```

Comment:
Perhaps use `clientSideIndexes`  here is better?

Reply:
Possibly but I have tested this code and do not want to go through the whole testing cycle which includes testing on a real cluster as this is a production issue for us now. Thanks for the suggestion